### PR TITLE
test(core): close RFC-64 codec invariant gaps

### PR DESCRIPTION
**Pull request by branarakic**
Focused canary replacement for the remaining useful pieces of superseded PRs #1795 and #1800.

What changes:
- preserves the existing extension-compatible `AuthorCatalogScopeV1` interface and keeps the shared author-lane validator as the single runtime owner of the paired governance invariant;
- proves both mixed governance permutations are rejected, including canonical wire parsing;
- pins every forbidden Unicode boundary and adjacent allowed value through both public catalog-coordinate and subgraph-name validators;
- adds the missing context-graph policy negatives for owner-signed unregistered governance, finalized contract mismatch, and digest-consistent unsigned and signed MemberRoster wrong-object-type envelopes in memory and on the wire.

Intentionally not carried:
- the #1795 private `iriComponentV1` change, because current canary imports that public helper from `catalog-seal-binding.ts`;
- the #1800 codec implementation, because canary already has the stronger discriminated publish-domain snapshot model.

Validation on exact head `f60f76cc2`:
- core dependency build: passed;
- focused codec tests: 37/37 passed;
- full core suite on the preceding behavior-equivalent head: 106 files, 1,691/1,691 tests passed;
- diff check against `testnet-canary`: passed.

Remaining-stack integration preview on exact `testnet-canary` `bba28b39e`:
- clean application in proposed order: #2341, #2340, #1823;
- dependency build through `@origintrail-official/dkg-agent`: passed;
- focused integrated suites: 254/254 passed (core 86, blockchain service 60, agent 108), including production native wiring;
- combined diff check: passed.

The initial head made `AuthorCatalogScopeV1` itself a union and exposed downstream TypeScript construction failures in CI. Commit `8c26782ff` restored the established interface; commit `4d5d4a0d5` removes the resulting orphan strict-governance export rather than advertising a static guarantee the public scope type does not provide.

Supersedes #1795 and #1800. No merge is requested by this PR creation.

### CODE REVIEW

File: packages/core/src/author-catalog-codec.ts [hunk 1]
```diff
@@ -97,6 +98,17 @@ const SCOPE_DOMAIN_BYTES = UTF8.encode(AUTHOR_CATALOG_SCOPE_DIGEST_DOMAIN_V1);
 const KEY_DOMAIN_BYTES = UTF8.encode(AUTHOR_CATALOG_KEY_DIGEST_DOMAIN_V1);
 const ROW_DOMAIN_BYTES = UTF8.encode(AUTHOR_CATALOG_ROW_DIGEST_DOMAIN_V1);
 
+/** Paired governance state committed by an author-catalog scope. */
+export type AuthorCatalogGovernanceV1 =
```
Comment (otReviewAgent) [inline packages/core/src/author-catalog-codec.ts]: 🟡 Issue: The new governance-pair abstraction is disconnected from the scope it describes

**What's wrong**
This adds a public type and an extra scalar import without using either to define a production boundary. Readers now see two competing representations of the same governance state: a precise union and the looser `AuthorCatalogScopeV1` actually consumed by APIs. That dead abstraction increases API surface while leaving callers dependent on a runtime-only invariant.

**Example**
A value with `governanceChainId: null` and a non-null `governanceContractAddress` still type-checks as `AuthorCatalogScopeV1`, even though the runtime assertion rejects it. The newly added union does not change that mismatch.

**Suggested direction**
Model this invariant once at its canonical owner—the shared author-lane scope, whose validator already enforces the pair—and derive catalog scopes from that model. If public interface compatibility prevents doing so in this PR, remove the orphan export rather than publishing a misleading abstraction.

**For Agents**
Rework the governance typing around `author-lane-scope-v1.ts` and `author-catalog-codec.ts`. Preserve the runtime paired-null invariant and any required interface compatibility, but either connect the union to the validated scope type/assertion boundary or remove it. Add a compile-time type check proving a mixed governance tuple is rejected if the invariant is represented statically.

Comment (branarakic) [reply packages/core/src/author-catalog-codec.ts]: Fixed in 4d5d4a0d5. I removed the orphan AuthorCatalogGovernanceV1 export and its now-unused ChainIdV1 import, so the public API no longer advertises a static guarantee that AuthorCatalogScopeV1 does not provide. The shared AuthorLaneScopeV1 runtime paired-null validation remains the canonical invariant boundary. Core build passed; the focused codec and policy suites passed 37/37; diff check passed.

File: packages/core/test/author-catalog-codec.test.ts [hunk 2]
```diff
@@ -110,6 +111,29 @@ describe('RFC-64 author catalog identifiers and graph names', () => {
...
+  it('pins every forbidden code-point boundary and its adjacent allowed values', () => {
+    for (const codePoint of [
+      0x0000, 0x001f,
+      0x007f, 0x009f,
+      0x00a0,
+      0x1680,
+      0x2000, 0x200b,
+      0x2028, 0x2029, 0x202f, 0x205f, 0x3000, 0xfeff,
+    ]) {
+      expect(isCatalogForbiddenCodePointV1(codePoint)).toBe(true);
```
Comment (otReviewAgent) [inline packages/core/test/author-catalog-codec.test.ts:124]: 🔴 Bug: The boundary test bypasses the graph-name validation path

**What's wrong**
The test claims to pin every forbidden boundary in the identifier/graph-name suite, but it checks only one predicate. A regression in the independent predicate used for subgraph names would not be detected.

**Example**
If U+1680 were accidentally removed from `isAuthorLaneForbiddenCodePointV1`, `assertSubGraphNameV1("a\u1680b")` would accept it while this new boundary test would remain green.

**Suggested direction**
Exercise the public identifier validators with strings containing each boundary value, particularly the separately implemented subgraph-name path.

**For Agents**
Update the boundary table in `author-catalog-codec.test.ts` to pass `String.fromCodePoint(codePoint)` through both `assertAssertionCoordinateV1` and `assertSubGraphNameV1`, asserting rejection for forbidden points and acceptance for adjacent allowed points. The direct predicate assertions may remain as supplemental API coverage.

File: packages/core/test/cg-policy-objects.test.ts [hunk 3]
```diff
@@ -433,6 +445,16 @@ describe('MemberRosterV1 codec', () => {
   it('enforces exact envelope type and signed digest', () => {
     expect(() => assertUnsignedMemberRosterEnvelopeV1(ROSTER_UNSIGNED)).not.toThrow();
     expect(() => assertSignedMemberRosterEnvelopeV1(ROSTER_SIGNED)).not.toThrow();
+    expect(() => assertUnsignedMemberRosterEnvelopeV1({
```
Comment (otReviewAgent) [inline packages/core/test/cg-policy-objects.test.ts:450]: 🔴 Bug: Exact object-type coverage omits the separate signed-envelope branch

**What's wrong**
The test name promises exact envelope-type enforcement alongside signed behavior, but all wrong-type mutations target only unsigned envelopes. The signed validator duplicates the object-type check and can regress independently.

**Example**
Removing the `assertObjectType` call from the signed member-roster validator would allow a signed roster carrying `objectType: "ContextGraphPolicyV1"`; all assertions added here would still pass because the only signed negative case changes the digest, not the type.

**Suggested direction**
Add a wrong-object-type regression case for the signed member-roster validator rather than relying on the unsigned checks and the unrelated signed-digest check.

**For Agents**
In the MemberRosterV1 envelope test, construct a valid signed roster with `objectType` changed to `CONTEXT_GRAPH_POLICY_OBJECT_TYPE_V1` and assert that `assertSignedMemberRosterEnvelopeV1` rejects it with `cg-policy-type`. Add the corresponding signed canonical-parser case if parser routing is also intended to be pinned.

File: packages/core/test/author-catalog-codec.test.ts (hunk 2 above)
Comment (branarakic) [reply packages/core/test/author-catalog-codec.test.ts:124]: Fixed in f60f76cc2. Every forbidden boundary is now passed through both public validators with String.fromCodePoint and asserted to reject, while every adjacent allowed value is asserted to pass both validators. The direct predicate assertions remain as supplemental coverage. This deliberately accepts either public rejection path because U+2000 is non-NFC and is correctly rejected by normalization before the forbidden-code-point check. Core build passed; the focused codec and policy suites passed 37/37.

File: packages/core/test/cg-policy-objects.test.ts (hunk 3 above)
Comment (branarakic) [reply packages/core/test/cg-policy-objects.test.ts:450]: Fixed in f60f76cc2. The signed wrong-type fixture now recomputes its generic control-object digest after changing objectType, so it passes signed-envelope digest validation and genuinely reaches the independent signed MemberRoster type check. Both the in-memory validator and canonical signed parser are asserted to reject with cg-policy-type. Core build passed; the focused codec and policy suites passed 37/37.

Comment (otReviewAgent) [reply packages/core/test/cg-policy-objects.test.ts:450]: 🟡 Issue: Reuse one wrong-type envelope instead of constructing it three ways

**What's wrong**
This block represents one scenario through three separate fixture-building mechanisms: two object literals and a textual replacement inside a canonical JSON fixture. That duplication makes the test unnecessarily indirect and forces future object-type or fixture changes to remain synchronized across several representations.

**Example**
Define one `wrongTypeUnsigned` fixture, pass it directly to the assertion, serialize it with `canonicalizeUnsignedControlEnvelopeBytes` for the unsigned parser, then derive `wrongTypeSigned` from that same object for the signed paths.

**Suggested direction**
Create one deliberately invalid unsigned envelope and serialize it through the generic unsigned control-envelope canonicalizer. Derive the signed fixture from the same object. This removes the duplicate object literals and brittle canonical-JSON replacement while making the invariant under test explicit.

**For Agents**
In `packages/core/test/cg-policy-objects.test.ts`, construct the wrong-object-type envelope once and reuse it across the unsigned assertion/parser and signed assertion/parser cases. Preserve all four rejection checks and run this test file afterward.
